src/lib.rs

```diff
@@ -155,10 +166,37 @@ pub fn init_with_rng<T: Rng>(rng: &mut T) -> Result<(), Error> {
 }
 
 /// Produces a 256-bit crypto hash out of the provided `data`.
-pub fn hash(data: &[u8]) -> [u8; 32] {
+pub fn hash(data: &[u8]) -> [u8; HASH_BYTES] {
     hashing_impl::sha3_256(data)
 }
 
+/// Uses Scrypt key derivation to securely derive a byte vector from a `password` and `salt`.
```

Comment:
Should we say Scrypt in the doc comment as the whole idea is to abstract stuff ? Maybe just call it password based key derivation or something.

Reply:
Good point 👍